tasks/unit_tests/omnibus_tests.py

```diff
@@ -175,3 +175,36 @@ def test_cache_is_disabled_by_unsetting_env_var(self):
         self.assertRunLines(['bundle exec omnibus build agent'])
         commands = _run_calls_to_string(self.mock_ctx.run.mock_calls)
         self.assertNotIn('omnibus-git-cache', commands)
+
+
+class TestOmnibusInstall(unittest.TestCase):
+    def setUp(self):
+        self.mock_ctx = MockContextRaising(run={})
+
+    def test_success(self):
+        self.mock_ctx.set_result_for('run', 'bundle install', Result())
+        omnibus.bundle_install_omnibus(self.mock_ctx)
+
+    def test_failure(self):
+        self.mock_ctx.set_result_for('run', 'bundle install', Result(exited=1))
+        try:
+            omnibus.bundle_install_omnibus(self.mock_ctx)
+            self.fail('Expected failure but got none')
+        except Exit:
+            pass
+        except Exception as e:
+            self.fail(f'Unexpected failure: {e}')
```

Comment:
If I'm not mistaken this is what you want to assert? ([docs ref](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises))
```suggestion
        with self.assertRaises(Exit):
            omnibus.bundle_install_omnibus(self.mock_ctx)
```

Reply:
Oh indeed that's much cleaner

Reply:
Remember that this probably also applies to  the `test_transient_repeated` test.